src/Request/uapi-request.js

```diff
@@ -45,6 +45,8 @@ module.exports = function uapiRequest(
     throw new RequestValidationError.ServiceUrlMissing();
   } else if (!auth || auth.username === undefined || auth.password === undefined) {
     throw new RequestValidationError.AuthDataMissing();
+  } else if (!auth.provider) {
+    auth.provider = '1G';
```

Comment:
We are checking for errors here.
So if provider will not be set, you will set it here, but this way you will stop checking for errors.

Please move it out of this if/else block.
And it would be even better to refactor this if/else block to separate if blocks